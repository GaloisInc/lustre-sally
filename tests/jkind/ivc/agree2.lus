-- There is no IVC for __GUARANTEE0

node _TOP__COMP(
  in1 : int;
  in2 : int;
  prop1 : bool;
  prop2 : bool;
  prop3 : bool;
  out1 : int;
  out2 : int
) returns (
  __ASSERT : bool
);
var
  __ASSUME__HIST : bool;
  __ASSUME__CONJ : bool;
  __COMP__PROP__0 : bool;
  __COMP__PROP__1 : bool;
  __COMP__PROP__2 : bool;
let
  __ASSUME__CONJ = true;

  __ASSUME__HIST = (__ASSUME__CONJ -> ((pre __ASSUME__HIST) and __ASSUME__CONJ));

  __COMP__PROP__0 = (out1 > (-1));

  __COMP__PROP__1 = (out1 < 1);

  __COMP__PROP__2 = (out2 > 10);

  __ASSERT = (__ASSUME__HIST => (__COMP__PROP__2 and (__COMP__PROP__1 and (__COMP__PROP__0 and true))));

  --%IVC __COMP__PROP__0,__COMP__PROP__1,__COMP__PROP__2;

tel;

node _TOP__COMP2(
  i1 : int;
  i2 : int;
  o1 : int;
  o2 : int
) returns (
  __ASSERT : bool
);
var
  __ASSUME__HIST : bool;
  __ASSUME__CONJ : bool;
  __COMP2__PROP__0 : bool;
  __COMP2__PROP__1 : bool;
let
  __ASSUME__CONJ = true;

  __ASSUME__HIST = (__ASSUME__CONJ -> ((pre __ASSUME__HIST) and __ASSUME__CONJ));

  __COMP2__PROP__0 = (o1 > 5);

  __COMP2__PROP__1 = (o2 > 9);

  __ASSERT = (__ASSUME__HIST => (__COMP2__PROP__1 and (__COMP2__PROP__0 and true)));

  --%IVC __COMP2__PROP__0,__COMP2__PROP__1;

tel;

node main(
  COMP__in1 : int;
  COMP__in2 : int;
  COMP___CLOCK_ : bool;
  COMP2__i1 : int;
  COMP2__i2 : int;
  COMP2___CLOCK_ : bool;
  input1 : int;
  input2 : int;
  COMP__prop1 : bool;
  COMP__prop2 : bool;
  COMP__prop3 : bool;
  COMP__out1 : int;
  COMP__out2 : int;
  COMP2__o1 : int;
  COMP2__o2 : int;
  output1 : int;
  output2 : int
) returns (

);
var
  __GUARANTEE0 : bool;
let
  --%MAIN;
  __GUARANTEE0 = ((input1 > 1) => (output1 = 5));

  assert (input1 = 0);

  assert condact(true, _TOP__COMP(COMP__in1, COMP__in2, COMP__prop1, COMP__prop2, COMP__prop3, COMP__out1, COMP__out2), true);

  assert condact(true, _TOP__COMP2(COMP2__i1, COMP2__i2, COMP2__o1, COMP2__o2), true);

  assert (input1 = COMP__in1);

  assert (input2 = COMP__in2);

  assert (input1 = COMP2__i1);

  assert (input2 = COMP2__i2);

  assert (COMP__out1 = output1);

  assert (COMP__out2 = output2);

  assert true;

  --%PROPERTY __GUARANTEE0;

tel;
