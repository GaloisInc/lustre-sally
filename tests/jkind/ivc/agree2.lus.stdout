Core Lustre
===========

node (COMP__in1 : int, COMP__in2 : int, COMP___CLOCK_ : bool, COMP2__i1 : int, COMP2__i2 : int, COMP2___CLOCK_ : bool, input1 : int, input2 : int, COMP__prop1 : bool, COMP__prop2 : bool, COMP__prop3 : bool, COMP__out1 : int, COMP__out2 : int, COMP2__o1 : int, COMP2__o2 : int, output1 : int, output2 : int)
  returns ()
  assumes (Eq, Eq$1, Eq$2, Eq$3, Eq$4, Eq$5, _TOP__COMP2, _TOP__COMP, Eq$6)
  shows (__GUARANTEE0)
let
  __GUARANTEE0 : bool
    = Implies(Gt(input1, 1), Eq(output1, 5))
   
  __ASSUME__CONJ$1 : bool
    = true
   
  pre___ASSUME__HIST$1 : bool
    = pre __ASSUME__HIST$1
   
  __ASSUME__HIST$1 : bool
    = __ASSUME__CONJ$1 -> And(pre___ASSUME__HIST$1, __ASSUME__CONJ$1)
   
  out1 : int
    = COMP__out1
   
  __COMP__PROP__0 : bool
    = Gt(out1, Neg(1))
   
  __COMP__PROP__1 : bool
    = Lt(out1, 1)
   
  out2 : int
    = COMP__out2
   
  __COMP__PROP__2 : bool
    = Gt(out2, 10)
   
  _TOP__COMP : bool
    = Implies(__ASSUME__HIST$1, And(__COMP__PROP__2, And(__COMP__PROP__1, And(__COMP__PROP__0, true))))
   
  __ASSUME__CONJ : bool
    = true
   
  pre___ASSUME__HIST : bool
    = pre __ASSUME__HIST
   
  __ASSUME__HIST : bool
    = __ASSUME__CONJ -> And(pre___ASSUME__HIST, __ASSUME__CONJ)
   
  o1 : int
    = COMP2__o1
   
  __COMP2__PROP__0 : bool
    = Gt(o1, 5)
   
  o2 : int
    = COMP2__o2
   
  __COMP2__PROP__1 : bool
    = Gt(o2, 9)
   
  _TOP__COMP2 : bool
    = Implies(__ASSUME__HIST, And(__COMP2__PROP__1, And(__COMP2__PROP__0, true)))
   
  in1 : int
    = COMP__in1
   
  in2 : int
    = COMP__in2
   
  prop1 : bool
    = COMP__prop1
   
  prop2 : bool
    = COMP__prop2
   
  prop3 : bool
    = COMP__prop3
   
  i1 : int
    = COMP2__i1
   
  i2 : int
    = COMP2__i2
   
  Eq$6 : bool
    = Eq(input1, 0)
   
  Eq$5 : bool
    = Eq(input1, COMP__in1)
   
  Eq$4 : bool
    = Eq(input2, COMP__in2)
   
  Eq$3 : bool
    = Eq(input1, COMP2__i1)
   
  Eq$2 : bool
    = Eq(input2, COMP2__i2)
   
  Eq$1 : bool
    = Eq(COMP__out1, output1)
   
  Eq : bool
    = Eq(COMP__out2, output2)
tel
Sally Model
===========

(define-state-type
   S
   (
     (Eq Bool)
     (i1 Int)
     (i2 Int)
     (in1 Int)
     (in2 Int)
     (input1 Int)
     (input2 Int)
     (o1 Int)
     (o2 Int)
     (out1 Int)
     (out2 Int)
     (output1 Int)
     (output2 Int)
     (prop1 Bool)
     (prop2 Bool)
     (prop3 Bool)
     (|COMP2___CLOCK_| Bool)
     (|COMP2__i1| Int)
     (|COMP2__i2| Int)
     (|COMP2__o1| Int)
     (|COMP2__o2| Int)
     (|COMP___CLOCK_| Bool)
     (|COMP__in1| Int)
     (|COMP__in2| Int)
     (|COMP__out1| Int)
     (|COMP__out2| Int)
     (|COMP__prop1| Bool)
     (|COMP__prop2| Bool)
     (|COMP__prop3| Bool)
     (|Eq:1| Bool)
     (|Eq:2| Bool)
     (|Eq:3| Bool)
     (|Eq:4| Bool)
     (|Eq:5| Bool)
     (|Eq:6| Bool)
     (|_TOP__COMP2| Bool)
     (|_TOP__COMP| Bool)
     (|__ASSUME__CONJ:1| Bool)
     (|__ASSUME__CONJ| Bool)
     (|__ASSUME__HIST:1:init| Bool)
     (|__ASSUME__HIST:1| Bool)
     (|__ASSUME__HIST:init| Bool)
     (|__ASSUME__HIST| Bool)
     (|__COMP2__PROP__0| Bool)
     (|__COMP2__PROP__1| Bool)
     (|__COMP__PROP__0| Bool)
     (|__COMP__PROP__1| Bool)
     (|__COMP__PROP__2| Bool)
     (|__GUARANTEE0| Bool)
     (|gal-initializing| Bool)
     (|pre___ASSUME__HIST:1| Bool)
     (|pre___ASSUME__HIST| Bool))
   (
     (input1 Int)
     (input2 Int)
     (output1 Int)
     (output2 Int)
     (|COMP2___CLOCK_| Bool)
     (|COMP2__i1| Int)
     (|COMP2__i2| Int)
     (|COMP2__o1| Int)
     (|COMP2__o2| Int)
     (|COMP___CLOCK_| Bool)
     (|COMP__in1| Int)
     (|COMP__in2| Int)
     (|COMP__out1| Int)
     (|COMP__out2| Int)
     (|COMP__prop1| Bool)
     (|COMP__prop2| Bool)
     (|COMP__prop3| Bool)))
(define-transition-system
   TS
   S
   (and
      (= |gal-initializing| true)
      (= |__ASSUME__HIST:1:init| true)
      (= |__ASSUME__HIST:init| true))
   (and
      (= next.|gal-initializing| false)
      (= next.|COMP__in1| input.|COMP__in1|)
      (= next.|COMP__in2| input.|COMP__in2|)
      (= next.|COMP___CLOCK_| input.|COMP___CLOCK_|)
      (= next.|COMP2__i1| input.|COMP2__i1|)
      (= next.|COMP2__i2| input.|COMP2__i2|)
      (= next.|COMP2___CLOCK_| input.|COMP2___CLOCK_|)
      (= next.input1 input.input1)
      (= next.input2 input.input2)
      (= next.|COMP__prop1| input.|COMP__prop1|)
      (= next.|COMP__prop2| input.|COMP__prop2|)
      (= next.|COMP__prop3| input.|COMP__prop3|)
      (= next.|COMP__out1| input.|COMP__out1|)
      (= next.|COMP__out2| input.|COMP__out2|)
      (= next.|COMP2__o1| input.|COMP2__o1|)
      (= next.|COMP2__o2| input.|COMP2__o2|)
      (= next.output1 input.output1)
      (= next.output2 input.output2)
      next.Eq
      next.|Eq:1|
      next.|Eq:2|
      next.|Eq:3|
      next.|Eq:4|
      next.|Eq:5|
      next.|_TOP__COMP2|
      next.|_TOP__COMP|
      next.|Eq:6|
      (=
         next.|__GUARANTEE0|
         (=>
            (< 1 next.input1)
            (= next.output1 5)))
      (= next.|__ASSUME__CONJ:1| true)
      (= next.|pre___ASSUME__HIST:1| state.|__ASSUME__HIST:1|)
      (ite
         state.|__ASSUME__HIST:1:init|
         (= next.|__ASSUME__HIST:1| next.|__ASSUME__CONJ:1|)
         (=
            next.|__ASSUME__HIST:1|
            (and next.|pre___ASSUME__HIST:1| next.|__ASSUME__CONJ:1|)))
      (= next.|__ASSUME__HIST:1:init| false)
      (= next.out1 next.|COMP__out1|)
      (=
         next.|__COMP__PROP__0|
         (<
            (- 1)
            next.out1))
      (=
         next.|__COMP__PROP__1|
         (< next.out1 1))
      (= next.out2 next.|COMP__out2|)
      (=
         next.|__COMP__PROP__2|
         (< 10 next.out2))
      (=
         next.|_TOP__COMP|
         (=>
            next.|__ASSUME__HIST:1|
            (and next.|__COMP__PROP__2| next.|__COMP__PROP__1| next.|__COMP__PROP__0| true)))
      (= next.|__ASSUME__CONJ| true)
      (= next.|pre___ASSUME__HIST| state.|__ASSUME__HIST|)
      (ite
         state.|__ASSUME__HIST:init|
         (= next.|__ASSUME__HIST| next.|__ASSUME__CONJ|)
         (=
            next.|__ASSUME__HIST|
            (and next.|pre___ASSUME__HIST| next.|__ASSUME__CONJ|)))
      (= next.|__ASSUME__HIST:init| false)
      (= next.o1 next.|COMP2__o1|)
      (=
         next.|__COMP2__PROP__0|
         (< 5 next.o1))
      (= next.o2 next.|COMP2__o2|)
      (=
         next.|__COMP2__PROP__1|
         (< 9 next.o2))
      (=
         next.|_TOP__COMP2|
         (=>
            next.|__ASSUME__HIST|
            (and next.|__COMP2__PROP__1| next.|__COMP2__PROP__0| true)))
      (= next.in1 next.|COMP__in1|)
      (= next.in2 next.|COMP__in2|)
      (= next.prop1 next.|COMP__prop1|)
      (= next.prop2 next.|COMP__prop2|)
      (= next.prop3 next.|COMP__prop3|)
      (= next.i1 next.|COMP2__i1|)
      (= next.i2 next.|COMP2__i2|)
      (=
         next.|Eq:6|
         (= next.input1 0))
      (=
         next.|Eq:5|
         (= next.input1 next.|COMP__in1|))
      (=
         next.|Eq:4|
         (= next.input2 next.|COMP__in2|))
      (=
         next.|Eq:3|
         (= next.input1 next.|COMP2__i1|))
      (=
         next.|Eq:2|
         (= next.input2 next.|COMP2__i2|))
      (=
         next.|Eq:1|
         (= next.|COMP__out1| next.output1))
      (=
         next.Eq
         (= next.|COMP__out2| next.output2))))

(query
   TS
   (or |gal-initializing| |__GUARANTEE0|))

[Lustre] Validating properties:
[Lustre] Property __GUARANTEE0... considering simultaneous states to depth 0
considering simultaneous states to depth 0
[Valid] considering simultaneous states to depth 0
[Lustre] Model status: [Valid] 
